src/Formula/reaction.ts

```diff
@@ -1,97 +1,114 @@
     )
     const res = input.enemy[`${ele}_resMulti`]
-    return infoMut(
-      // CAUTION:
-      // Add amp multiplier/additive term only to swirls that have amp/additive reactions.
-      // It is wasteful to add them indiscriminately, but this means
-      // that we need to audit and add appropriate elements here
-      // should amp/additive reactions be added to more swirls.
-      ["pyro", "hydro", "cryo", "electro"].includes(ele)
-        ? (ele === "electro"
-          // Additive reactions apply the additive term before resistance, but after swirl bonuses
-          ? data(prod(sum(base, input.hit.addTerm), res), { hit: { ele: constant(ele) } })
-          // Amp reaction
-          : data(prod(base, res, input.hit.ampMulti), { hit: { ele: constant(ele) } }))
-        : prod(base, res),
-      { key: `${ele}_swirl_hit` })
+    // CAUTION:
+    // Add amp multiplier/additive term only to swirls that have amp/additive reactions.
+    // It is wasteful to add them indiscriminately, but this means
+    // that we need to audit and add appropriate elements here
+    // should amp/additive reactions be added to more swirls.
+    return ["pyro", "hydro", "cryo", "electro"].includes(ele)
+      ? (ele === "electro"
+        // Additive reactions apply the additive term before resistance, but after swirl bonuses
+        ? data(prod(sum(base, input.hit.addTerm), res), { hit: { ele: constant(ele) } })
+        // Amp reaction
+        : data(prod(base, res, input.hit.ampMulti), { hit: { ele: constant(ele) } }))
+      : prod(base, res)
   })
 }
-export const reactions = {
-  anemo: {
-    electroSwirl: trans.swirl.electro,
-    pyroSwirl: trans.swirl.pyro,
-    cryoSwirl: trans.swirl.cryo,
-    hydroSwirl: trans.swirl.hydro,
-    overloaded: trans.overloaded,
-    electrocharged: trans.electrocharged,
-    superconduct: trans.superconduct,
-    shattered: trans.shattered,
-    burning: trans.burning,
-    bloom: trans.bloom,
-    burgeon: trans.burgeon,
-    hyperbloom: trans.hyperbloom,
-  },
-  geo: {
-    crystallize: crystallizeHit,
-    ...Object.fromEntries(absorbableEle.map(e => [`${e}Crystallize`,
-    infoMut(prod(percent(2.5), crystallizeHit), { key: `${e}_crystallize`, variant: e })])),
-    shattered: trans.shattered,
-  },
-  electro: {
-    overloaded: trans.overloaded,
-    electrocharged: trans.electrocharged,
-    superconduct: trans.superconduct,
-    shattered: trans.shattered,
-    hyperbloom: trans.hyperbloom,
-  },
-  hydro: {
-    electrocharged: trans.electrocharged,
-    shattered: trans.shattered,
-    bloom: trans.bloom,
-  },
-  pyro: {
-    overloaded: trans.overloaded,
-    shattered: trans.shattered,
-    burning: trans.burning,
-    burgeon: trans.burgeon,
-  },
-  cryo: {
-    superconduct: trans.superconduct,
-    shattered: trans.shattered,
-  },
-  dendro: {
-    burning: trans.burning,
-    bloom: trans.bloom,
+export function getReactions(charElement: ElementKey) {
```

Comment:
My opinion is that we should make reactions here pretty much static object, instead of a generation function. construct reactions by reading `input.charEle` and `input.infusion` should be enough?